pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1442,3 +1442,40 @@ def test_merge_series(on, left_on, right_on, left_index, right_index, nm):
         with pytest.raises(ValueError, match='a Series without a name'):
             result = pd.merge(a, b, on=on, left_on=left_on, right_on=right_on,
                               left_index=left_index, right_index=right_index)
+
+
+@pytest.mark.parametrize("col1, col2, suffixes, expected_cols", [
+    (0, 0, ("", "_dup"), ["0", "0_dup"]),
+    (0, 0, (None, "_dup"), [0, "0_dup"]),
+    (0, 0, ("_x", "_y"), ["0_x", "0_y"]),
+    ("a", 0, (None, "_y"), ["a", 0]),
+    (0.0, 0.0, ("_x", None), ["0.0_x", 0.0]),
+    ("b", "b", (None, "_y"), ["b", "b_y"]),
+    ("a", "a", ("_x", None), ["a_x", "a"]),
+    ("a", "b", ("_x", None), ["a", "b"])
+])
+def test_merge_suffix(col1, col2, suffixes, expected_cols):
+    # issue: 24782
+    a = pd.DataFrame({col1: [1, 2, 3]})
+    b = pd.DataFrame({col2: [4, 5, 6]})
+
+    expected = pd.DataFrame([[1, 4], [2, 5], [3, 6]],
+                            columns=expected_cols)
+
+    result = a.merge(b, left_index=True, right_index=True, suffixes=suffixes)
```

Comment:
many tests in `test_merge.py` repeat using both the `left.merge(right, **kwargs)` and the `pd.merge(left, right, **kwargs)` syntax. 

i'm not sure how important that is, so unless one of the maintainers comments otherwise this lgtm

Reply:
thanks for your feedback, @simonjayhawkins ! if one of maintainers brings this up and wants it to be standardised in this PR, i will then fix it! otherwise, as you said, I will open another PR to tackle it!